### PR TITLE
[bugfix] DeepSpeed with no schedulers

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

On master:
Screenshot 2021-07-27 at 15 58 09

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]
great fix, thanks @tchaton!

**Review by ethanwharris — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: tests/plugins/test_deepspeed_plugin.py [hunk 1]
```diff
@@ -687,3 +692,17 @@ def _assert_save_model_is_equal(model, tmpdir, trainer, cls=BoringModel):
         # Assert model parameters are identical after loading
         for orig_param, trained_model_param in zip(model.parameters(), saved_model.parameters()):
             assert torch.equal(orig_param, trained_model_param)
+
+
+@RunIf(min_gpus=2, deepspeed=True, special=True)
+def test_deepspeed_multigpu_no_schedulers(tmpdir):
+    """
+    Test to ensure ZeRO Stage 3 works with a parallel model.
```
Comment (awaelchli) [inline tests/plugins/test_deepspeed_plugin.py:700]:
```suggestion
    Test to ensure ZeRO Stage 3 works with a parallel model and no schedulers.
```

**Review by awaelchli — APPROVED** [review]
LGTM

the chlog will have to move up after the recent 1.4 release :)

**Review by Borda — APPROVED** [review]

**Comment by kaushikb11** [comment]
>the chlog will have to move up after the recent 1.4 release :)

I haven't published the release yet. Will add this as well.

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 2]
```diff
@@ -430,8 +431,9 @@ def _initialize_deepspeed_train(self, model):
 
         # although we set these here, deepspeed manages the specific optimizer logic
         self.lightning_module.trainer.optimizers = [deepspeed_optimizer]
-        lr_scheduler["scheduler"] = deepspeed_scheduler
-        self.lightning_module.trainer.lr_schedulers = [lr_scheduler]
+        if scheduler:
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/deepspeed.py:434]: should this be?:

```suggestion
        if deepspeed_scheduler is not None:
```

we check that `"optimizer" in self.config` but not the scheduler key.

Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/deepspeed.py:434]: I think you are correct @carmocca as this ensures that if the user defines a scheduler within the config object, this is referenced lr_scheduler

Comment (SeanNaren) [reply pytorch_lightning/plugins/training_type/deepspeed.py]: Another test should be added that ensures if a user passes a scheduler via the deepspeed config, this is correctly set

**Review by carmocca — APPROVED** [review]
